File: backend/src/main/java/ca/metricalsky/winston/entity/VideoRecordingLocationEntity.java
package ca.metricalsky.winston.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "video_recording_locations")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VideoRecordingLocationEntity {

    @Id
    @Column(name = "video_id")
    private String videoId;

    @Column(name = "description")
    private String description;

    @Column(name = "latitude")
    private Double latitude;

    @Column(name = "longitude")
    private Double longitude;

    @Column(name = "altitude")
    private Double altitude;
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/entity/video/VideoRecordingLocationEntity.kt
package ca.metricalsky.winston.database.entity.video

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "video_recording_locations")
class VideoRecordingLocationEntity(

    @Id
    @Column(name = "video_id")
    var videoId: String,

    @Column(name = "description")
    var description: String?,

    @Column(name = "latitude")
    var latitude: Double?,

    @Column(name = "longitude")
    var longitude: Double?,

    @Column(name = "altitude")
    var altitude: Double?,
)
